src/ethereum/london/vm/interpreter.py

```diff
@@ -107,16 +105,19 @@ def process_message_call(
             evm = process_create_message(message, env)
     else:
         evm = process_message(message, env)
-
-    accounts_to_delete = collect_accounts_to_delete(evm)
-    refund_counter = U256(calculate_gas_refund(evm))
+        if account_exists_and_is_empty(env.state, Address(message.target)):
+            evm.touched_accounts.add(Address(message.target))
 
     return MessageCallOutput(
         gas_left=evm.gas_left,
-        refund_counter=refund_counter,
+        refund_counter=U256(evm.refund_counter)
+        if not evm.has_erred
+        else U256(0),
         logs=evm.logs if not evm.has_erred else (),
-        accounts_to_delete=accounts_to_delete,
-        touched_accounts=collect_touched_accounts(evm),
+        accounts_to_delete=evm.accounts_to_delete
+        if not evm.has_erred
+        else set(),
+        touched_accounts=evm.touched_accounts if not evm.has_erred else set(),
```

Comment:
These conditionals are maybe too Pythonic? Plus they're definitely too hard to read :rofl: